### PR TITLE
Fixes #17 : Fixes when set -f option, no set -f option automatically

### LINKED ISSUE #17: When set -f option, no set -f option automatically
`Zend\Mail\Transport\SendMail` always append envelope-from parameter.  
But I want to set any envelope-from.  
In this PR, only when not set option `-f`, set `-f` option automatically.
This means if I set `-ffoo@example.com` or `-bs -ffoo@example.com`, `Zend\Mail\Transport\Sendmail` doesn't set `-f` option.

---

Originally posted by @tzmfreedom at https://github.com/zendframework/zend-mail/pull/227

### PR DESCRIPTION
**Pull request by samsonasik**
|    Q          |   A
|-------------- | ------
| Bugfix        | yes

### Description

Re-create https://github.com/zendframework/zend-mail/pull/227 by @tzmfreedom Fixes #17 .  Original PR description : 

```
Zend\Mail\Transport\SendMail always append envelope-from parameter.
But I want to set any envelope-from.
In this PR, only when not set option -f, set -f option automatically.
This means if I set -ffoo@example.com or -bs -ffoo@example.com, Zend\Mail\Transport\Sendmail doesn't set -f option.
```

### CODE REVIEW

File: src/Transport/Sendmail.php [hunk 1]
```diff
@@ -260,6 +260,9 @@ protected function prepareParameters(Mail\Message $message)
         }
 
         $parameters = (string) $this->parameters;
+        if (preg_match('/^\-f.+$/', $parameters) || strpos($parameters, ' -f') !== false) {
```
Comment (weierophinney) [inline src/Transport/Sendmail.php]:
```suggestion
        if (strpos($parameters, '-f') !== false) {
```

This accomplishes the same thing, with one less operation.

**Review by weierophinney — CHANGES_REQUESTED** [review]
I've made one suggestion to how you check for the `-f` flag.

Additionally, we need tests demonstrating that the `setParameters()` method uses the parameter passed to the transport and NOT the sender and/or from address when it is provided; currently, you only demonstrate it is used when provided, and not used when not present, but not the additional case where it is not provided and falls through to the sender or from address.

File: src/Transport/Sendmail.php (hunk 1 above)
Comment (michalbundyra) [reply src/Transport/Sendmail.php]: @weierophinney it doesn't seem right, imo

it's not the same, previously we accept `-f` at the beginning of when it is after a space.

I can imagine case `my-fancy-param` where `-f` is no at the beginning nor after a space.

Comment (weierophinney) [reply src/Transport/Sendmail.php]: Then do it as follows:

```php
if (preg_match('/(^| )\-f.+/', $parameters)) {
```

It doesn't make sense to have two conditions when it's essentially the same condition ("is there a -f switch provided?"). The above regex does the "at the beginning" as well as "following a space" conditions.

Comment (michalbundyra) [reply src/Transport/Sendmail.php]: that's better ;-) there is also redundant `\` before `-` in the regexp

Comment (samsonasik) [reply src/Transport/Sendmail.php]: @weierophinney updated.

**Review by weierophinney — APPROVED** [review]
I've combined the two unit tests into a single one backed by a data provider. I think it's ready to :ship: once tests have run.

**Pull request merged**